lib/service_order_listener.rb

```diff
@@ -0,0 +1,44 @@
+class ServiceOrderListener
+  attr_accessor :messaging_client_options, :client
+
+  def initialize(messaging_client_options = {})
+    self.messaging_client_options = default_messaing_options.merge(messaging_client_options)
+  end
+
+  def run
+    Thread.new do
+      self.client = ManageIQ::Messaging::Client.open(messaging_client_options)
+
+      client.subscribe_messages({
+        :service   => "platform.topological-inventory.task-output-stream",
```

Comment:
@eclarizio Is this the unique consumer group that prevents multiple threads from picking up the same work?